configuration/configuration_test.go

```diff
@@ -0,0 +1,153 @@
+package configuration_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/almighty/almighty-core/configuration"
+	"github.com/almighty/almighty-core/resource"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetConfigurationDataSucess(t *testing.T) {
+
+	// this, in reality gets set in main
+	os.Setenv("ALMIGHTY_CONFIG_FILE_PATH", "../config.yaml")
+	cd, err := configuration.GetConfigurationData()
+	assert.Nil(t, err)
+	assert.NotNil(t, cd)
+}
+
+func TestNewConfigurationDataSuccess(t *testing.T) {
+
+	resource.Require(t, resource.UnitTest)
+	configFilePath := "../config.yaml"
+	cd, err := configuration.NewConfigurationData(configFilePath)
+	assert.Nil(t, err)
+	assert.NotNil(t, cd)
+
+}
+
+func TestNewConfigurationDataFail(t *testing.T) {
+
+	resource.Require(t, resource.UnitTest)
+	configFilePath := "../invalid_config.yaml"
+	_, err := configuration.NewConfigurationData(configFilePath)
+	assert.NotNil(t, err)
+
+}
+
+func TestGetKeycloakEndpointToken(t *testing.T) {
+
+	cd := getConfigurationDataHandler()
+
+	t.Log(cd.GetKeycloakEndpointToken())
+	assert.NotNil(t, cd.GetKeycloakEndpointToken())
+
+}
+
+func TestGetPostgresHost(t *testing.T) {
+	configurationData := getConfigurationDataHandler()
+	assert.NotNil(t, configurationData.GetPostgresHost())
```

Comment:
Thanks @kwk ,

> Why do you check for NotNil when GetPostgresHost returns a string value? Please also double check the other values. 
I actually something similar in one place and thought perhaps I'm over-doing stuff - good to know it's not. :P 

`assert.Equal(t, configurationData.GetPostgresConnectionMaxRetries(), 50)`